src/game/game_text.hpp

```diff
@@ -29,12 +29,12 @@ struct LanguageStrings {
 
 /** Container for all the game strings. */
 struct GameStrings {
-	uint version;                  ///< The version of the language strings.
-	LanguageStrings *cur_language; ///< The current (compiled) language.
+	uint version;                                  ///< The version of the language strings.
+	std::shared_ptr<LanguageStrings> cur_language; ///< The current (compiled) language.
 
-	AutoDeleteSmallVector<LanguageStrings *> raw_strings;      ///< The raw strings per language, first must be English/the master language!.
-	AutoDeleteSmallVector<LanguageStrings *> compiled_strings; ///< The compiled strings per language, first must be English/the master language!.
-	StringList string_names;                                   ///< The names of the compiled strings.
+	std::vector<std::unique_ptr<LanguageStrings>> raw_strings;      ///< The raw strings per language, first must be English/the master language!.
+	std::vector<std::shared_ptr<LanguageStrings>> compiled_strings; ///< The compiled strings per language, first must be English/the master language!.
```

Comment:
how difficult would it be to "fix" the use of shared_ptr here?
or, perhaps the various places that "own" the pointer could be documented?

Reply:
Very easily, it is only `GameStrings::cur_language` that points to one of the compiled string entries. We store a raw pointer now and just assume it's never going to be invalidated; still possible.

In pure though, raw pointers are bad and should never be used. Of course, the codebase is far, far, far from that point now. So, philosophical question time: do the "right" thing, or the pragmatic thing?